### PR TITLE
Move Mounted Files to .aws-sam Folder

### PR DESCRIPTION
**Pull request by awood45**
*Issue #, if available:* Resolves issue #2095 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samcli/commands/local/cli_common/options.py [hunk 1]
```diff
@@ -117,6 +145,20 @@ def invoke_common_options(f):
                 help="Specifies the location basedir where the Layers your template uses will be downloaded to.",
                 default=get_default_layer_cache_dir(),
             ),
+            click.option(
+                "--rapid-basedir",
```
Comment (c2tarun) [inline samcli/commands/local/cli_common/options.py:149]: Do we want to expose this functionality?

File: samcli/local/docker/lambda_container.py [hunk 2]
```diff
@@ -143,20 +144,20 @@ def _get_additional_options(runtime, debug_options):
         return opts
 
     @staticmethod
-    def _get_additional_volumes(runtime, debug_options):
+    def _get_additional_volumes(runtime, debug_options, mounted_file_provider):
```
Comment (c2tarun) [inline samcli/local/docker/lambda_container.py:147]: missing in doc

File: samcli/local/mount/mounted_file_provider.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        return self._rapid_basedir
+
+    @property
+    def go_bootstrap_basedir(self):
+        return self._go_bootstrap_basedir
+
+    @staticmethod
+    def _copy_source_files(rapid_basedir, go_bootstrap_basedir):
+        LOG.debug("Creating basedirs if they do not yet exist.")
+        Path(rapid_basedir).mkdir(mode=0o700, parents=True, exist_ok=True)
```
Comment (c2tarun) [inline samcli/local/mount/mounted_file_provider.py:36]: I am assuming that python somehow handles this for Windows.

File: samcli/local/mount/mounted_file_provider.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        rapid_dest = "{}/init".format(rapid_basedir)
+        LOG.debug("Copying RAPID stub server from %s to %s.", str(MountedFileProvider._RAPID_SOURCE), rapid_dest)
+        copyfile(MountedFileProvider._RAPID_SOURCE, rapid_dest)
+        rapid_st = os.stat(MountedFileProvider._RAPID_SOURCE)
+        os.chmod(rapid_dest, rapid_st.st_mode)
+
+        go_bootstrap_dest = "{}/aws-lambda-go".format(go_bootstrap_basedir)
+        copyfile(MountedFileProvider._GO_BOOTSTRAP_SOURCE, go_bootstrap_dest)
+        go_bootstrap_st = os.stat(MountedFileProvider._GO_BOOTSTRAP_SOURCE)
+        os.chmod(go_bootstrap_dest, go_bootstrap_st.st_mode)
```
Comment (c2tarun) [inline samcli/local/mount/mounted_file_provider.py:48]: nitpick: probably can be extracted in a method and called for both files.

File: samcli/local/mount/mounted_file_provider.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import os
+import stat
+
+from pathlib import Path
+from shutil import copyfile
+
+LOG = logging.getLogger(__name__)
+
+
+class MountedFileProvider:
```
Comment (c2tarun) [inline samcli/local/mount/mounted_file_provider.py:14]: Some doc would be helpful here when we'll see this in future 😄

File: samcli/commands/local/cli_common/options.py (hunk 1 above)
Comment (awood45) [reply samcli/commands/local/cli_common/options.py:149]: Will look at an alternative approach tomorrow, but short answer is "yes" - if the user has folder permission issues and we don't provide this option, they're basically stuck with no ability to invoke. However, we should carefully document that most users will never want to touch this.

File: samcli/local/mount/mounted_file_provider.py (hunk 3 above)
Comment (awood45) [reply samcli/local/mount/mounted_file_provider.py:36]: It does, we use this for global config and layer caching.

**Comment by awood45** [comment]
We may pick this back up later if needed, but closing in favor of #2100

**Pull request closed without merging**